### DISCUSSION TITLE
3.7.1 -> make install -> [doc/ccache.1] Error 1

### DISCUSSION THREAD

**Discussion by haiomaio2** [ccache/ccache#425] [Support]
### Question ###

Is this is a bug, or have I forgotten /am I missing sth /package ?

make install
  A2X      doc/ccache.1
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/etc/asciidoc/docbook-xsl/manpage.xsl" "/home/user/Downloads/ccache/doc/MANUAL.xml" returned non-zero exit status 6
dev.mk:195: recipe for target 'doc/ccache.1' failed
make: *** [doc/ccache.1] Error 1

MANUAL.xml is empty

**Comment by jrosdahl** [comment]
As noted in [`doc/INSTALL.md`](https://github.com/ccache/ccache/blob/master/doc/INSTALL.md), you need AsciiDoc if you build directly from a source repository. But the recommended way is to build from a [release archive](https://github.com/ccache/ccache/blob/master/doc/INSTALL-from-release-archive.md) – then you only need a C compiler installed.

**Comment by haiomaio2** [comment]
AsciiDoc had been installed.

$ asciidoc --version
asciidoc 8.6.10

**Comment by jrosdahl** [comment]
OK, then it looks like [xsltproc](http://xmlsoft.org/XSLT/xsltproc2.html) is missing. I've improved INSTALL.md to mention it in 974c4eb5ef10.

**Comment by haiomaio2** [comment]
Unfortunately. not that easy. xsltproc has been installed
```
$ xsltproc --version
Using libxml 20904, libxslt 10129 and libexslt 817
xsltproc was compiled against libxml 20904, libxslt 10129 and libexslt 817
libxslt 10129 was compiled against libxml 20904
libexslt 817 was compiled against libxml 20904
```
Isnt it because of ?
`MANUAL.xml is empty`

Is "dynamically linked, interpreter /lib64/l" meant to be like this ?

```
$ file /usr/local/bin/ccache
/usr/local/bin/ccache: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/l, for GNU/Linux 3.2.0, BuildID[sha1]=f90ffb757a38a9687a9e27c0248c7ed7d0de9ac7, with debug_info, not stripped
```

Further
```
$ file ccache 
ccache: cannot open `ccache' (No such file or directory)
```
and
```
$ ccache --version
ccache version 3.7.1
```

**Comment by jrosdahl** [comment]
> Isnt it because of ?
> MANUAL.xml is empty

MANUAL.xml is a temporary file that likely became empty since a2x couldn't run xsltproc. Just remove it or clean everything and start from the beginning.

**Comment by haiomaio2** [comment]
It was a clean build ... the error comes always. It might be some other a2x releated issue. Have you tried it with a clean build as well?

**Comment by jrosdahl** [comment]
> Have you tried it with a clean build as well?

Yes, this works for me:

```
% git clone https://github.com/ccache/ccache.git
[...]
% cd ccache
% ./autogen.sh
Now run ./configure and make
% ./configure
[...]
% make install V=1
[...]
asciidoc -a revnumber=3.7.1+26_g974c4eb5ef10 -d manpage -b docbook -o - doc/MANUAL.adoc | \
  perl -pe 's!<literal>(.*?)</literal>!<emphasis role="strong">\1</emphasis>!g' >doc/MANUAL.xml
a2x --doctype manpage --format manpage doc/MANUAL.xml
cp doc/ccache.1 ccache.1
[...]
```

Try running `make install V=1` to print the actual commands that are being run. There should be something similar to what I indicated above in the output.

**Comment by haiomaio2** [comment]
For some reason, I cannot reproduce the error anymore (although Im using the same tar.gz as before).
The only difference is, I updated the ubuntu ...

Thanks anyway once again.

**Comment by jrosdahl** [comment]
> Im using the same tar.gz as before

Wait a minute, are you building from a release archive (tar.gz), not from a Git repository? If so, there should be no need in the first place to install asciidoc or xsltproc.

**Comment by jrosdahl** [comment]
@haiomaio2: Any input to my question above? (It's not a rhethorical question; I want to understand if there are any issues with the release archives.)

**Comment by jrosdahl** [comment]
@haiomaio2: I'm still interested in an answer to my question.

**Comment by tamboril** [comment]
I'm running into this right now, building from a1964c685b.  doc/MANUAL.xml is 0 bytes

**Comment by AlexanderLanin** [comment]
Could you check the versions as mentioned above? And run: make install V=1

**Comment by tamboril** [comment]
```
asciidoc 8.6.8

xsltproc --version
Using libxml 20901, libxslt 10128 and libexslt 817
xsltproc was compiled against libxml 20901, libxslt 10128 and libexslt 817
libxslt 10128 was compiled against libxml 20901
libexslt 817 was compiled against libxml 20901

jharris@dev26 (master)|~/repos/ccache> make install V=1
a2x --doctype manpage --format manpage doc/MANUAL.xml
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/etc/asciidoc/docbook-xsl/manpage.xsl" "/home/jharris/repos/ccache/doc/MANUAL.xml" returned non-zero exit status 6
make: *** [dev.mk:176: doc/ccache.1] Error 1
```

**Comment by jrosdahl** [comment]
> doc/MANUAL.xml is 0 bytes

That will happen if you (a) don't have asciidoc installed, (b) build ccache, (c) get a build error about missing asciidoc, (d) install asciidoc and (e) build again in the previous build tree. Is this what happened? If so, start again from a clean tree or just delete doc/MANUAL.xml and try again.

**Comment by tamboril** [comment]
That was indeed what happened.  I removed doc/MANUAL.xml and did a make
install again with no issues.

Thanks

On Sun, May 24, 2020 at 6:51 AM Joel Rosdahl <notifications@github.com>
wrote:

> doc/MANUAL.xml is 0 bytes
>
> That will happen if you (a) don't have asciidoc installed, (b) build
> ccache, (c) get a build error about missing asciidoc, (d) install asciidoc
> and (e) build again in the previous build tree. Is this what happened? If
> so, start again from a clean tree or just delete doc/MANUAL.xml and try
> again.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ccache/ccache/issues/425#issuecomment-633219477>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD4B47JK43HHRIQY3AZM5ZTRTEC5BANCNFSM4HNTKUXQ>
> .
>


-- 
John Harris
john@johnharrisweb.me

**Comment by Sanaki** [comment]
In case some other poor soul finds themselves here with the same issue, I encountered this just now due to a mix of ccache speeding certain stages up significantly and multi-threaded compilation (`make -j5`) outpacing necessary files. Soon as I ran `make` bare, the issue resolved.
